src/app/register/register.component.ts

```diff
@@ -273,7 +273,7 @@ export class RegisterComponent implements OnInit, OnDestroy, AfterViewInit {
   }
 
   get readyToTakeAccountDetails(): boolean {
-    return !!this.verificationCodeSupplied || this.emailVerificationToken?.valid || ! flags.requireEmailVerification
+    return !!this.verificationCodeSupplied || !!this.emailVerificationToken?.valid
```

Comment:
Curious whether there's a particular reason this needs the extra `!!` now? Seems like `this.emailVerificationToken` is typed strictly enough for it to be safe without, but I guess there's value for quick reading in making it similar to the left hand side?

Reply:
Without the extra `!!` there's a type error `Type boolean | undefined is not assignable to type boolean`

Previously it wasn't needed since on the left of a `||` undefined acts as false.